core/go/internal/plugins/plugin_base_test.go

```diff
@@ -94,7 +95,9 @@ func (tp *mockPlugin[T]) Run(grpcTarget, pluginId string) {
 		if tp.sendRequest != nil {
 			req := tp.sendRequest(pluginId)
 			err := stream.Send(req)
-			require.NoError(t, err)
+			if tp.expectClose == nil {
+				require.NoError(t, err)
```

Comment:
Hypothesis is that one of these `require` statements is exiting the thread without calling `expectClose()`